### PR TITLE
Fix Windows embedded file path bug

### LINKED ISSUE #102: Embedded file system file path creation fails on Windows
**Description**

When creating the path to [embedded files](https://pkg.go.dev/embed), the code uses the `filepath.Join`, which will use a back slash as the path separator on the Windows platform. However, the embedded file system always uses forward slashes as the path separator, even on Windows, see the docs [here](https://pkg.go.dev/embed#hdr-Directives).

**Version**

v0.2.0

### PR DESCRIPTION
**Pull request by malancas**
#### Summary

This fixes a Windows compatibility issue related to embedded file paths and a bug downloading target metadata in the go-tuf library.

Fixes https://github.com/sigstore/sigstore-go/issues/102

### CODE REVIEW

**Review by kommendorkapten — DISMISSED** [review]

**Review by Hayden-IO — COMMENTED** [review]
Build action is stuck

**Comment by malancas** [comment]
> Build action is stuck

I'll take a look

**Comment by kommendorkapten** [comment]
On the stuck build job @malancas wrote this to me:

> I see the build workflow is hanging on the PR, I'm guessing it has to do with the new matrix jobs and the build job not matching the required checks list anymore

And the branch protection rule requires a job named `build` to run https://github.com/sigstore/community/blob/main/github-sync/github-data/sigstore/repositories.yaml#L1860
And we can see that in the list above.

Can that be the issue, as it does say: _1 expected check_, but there is no link to any details. The build jobs from the build with a matrix are reported as "Build and test/build (<platform>)".

How about renaming the `build` job to `build-and-test` then create a new job `build`, that does nothing but _requires_ all the build from `build-and-test`?

**Comment by malancas** [comment]
The required cosign status checks account for platform related matrix jobs: https://github.com/sigstore/community/blob/main/github-sync/github-data/sigstore/repositories.yaml#L228-L230. We could update the sigstore-go required checks entry to match this.

**Comment by Hayden-IO** [comment]
Ah good catch, yea we'll need to update the checks in the community repo.

**Comment by malancas** [comment]
I'll open a PR for that now

File: go.mod
Comment (Hayden-IO) [inline go.mod:19]: @kommendorkapten not a blocker, but do you know when go-tuf maintainers will be cutting a release with the v2 implementation?

**Comment by malancas** [comment]
When someone has time to review: https://github.com/sigstore/community/pull/408

**Review by Hayden-IO — APPROVED** [review]
Should be good now. Once merged we'll just need to rebase other PRs

**Review by Hayden-IO — APPROVED** [review]

File: go.mod
Comment (kommendorkapten) [reply go.mod:19]: @haydentherapper there is still one open bug for Windows in go-tuf that I intend to fix today. When all test cases are working for all platforms, and we have a clear understanding of the current open issues, I think it's time we create a first release. I would guess one-two weeks from now?

**Review by alphachart — COMMENTED** [review]
.

**Pull request merged**
